cli/npm/managed/registry.rs

```diff
@@ -148,9 +148,11 @@ impl CliNpmRegistryApiInner {
             || !self.previously_reloaded_packages.lock().insert(name.to_string())
           {
             // attempt to load from the file cache
-            if let Some(info) = self.load_file_cached_package_info(name) {
+            if let Some(info) = self.load_file_cached_package_info(name).await {
```

Comment:
I don't think we can await this here because then multiple threads will run into this code. Instead of this, can you lift the code from my pr here and combine it into this one? We noticed the same thing at the same time https://github.com/denoland/deno/pull/24019

Reply:
Yup was just realizing that. I went ahead and dropped this commit and cherrypicked yours instead